src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy

```diff
@@ -247,6 +254,46 @@ class BidValidationSpec extends BaseSpec {
         assert response.ext?.warnings[ErrorType.PREBID]*.message ==
                 ["Dropped bid '$invalidBid.id'. Does not contain a positive (or zero if there is a deal) 'price'" as String]
 
+        where:
+        debug | test | bidPrice                      | dealId
+        0     | 1    | PBSUtils.randomNegativeNumber | null
+        0     | 1    | PBSUtils.randomNegativeNumber | PBSUtils.randomNumber
+        0     | 1    | 0                             | null
+        0     | 1    | null                          | PBSUtils.randomNumber
+        0     | 1    | null                          | null
+        1     | 0    | PBSUtils.randomNegativeNumber | null
+        1     | 0    | PBSUtils.randomNegativeNumber | PBSUtils.randomNumber
+        1     | 0    | 0                             | null
+        1     | 0    | null                          | PBSUtils.randomNumber
+        1     | 0    | null                          | null
+    }
+
+    def "PBS should drop invalid bid without debug error when request debug disabled and bid price is #bidPrice and deal id is #dealId"() {
+        given: "Default basic BidRequest with generic bidder"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            test = 0
+            ext.prebid.debug = 0
+        }
+
+        and: "Bid response"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest)
+        def bid = bidResponse.seatbid.first().bid.first()
+        bid.dealid = dealId
+        bid.price = bidPrice
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse)
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Invalid bid should be deleted"
+        assert response.seatbid.size() == 0
+
```

Comment:
Strangely, seat bid is not about invalid bids, better to check the seatNonBid field.